feats/app.py

```diff
@@ -53,6 +53,9 @@ def get_current_state(self) -> FeatureState:
         except IndexError:
             return None
 
+        if state_data is None:
```

Comment:
Ah, looks like I"m not returning the correct thing from the last PR. I'll update this to raise an IndexError in first()/last() instead of returning None